### PR TITLE
{Identity} Fix logout

### PR DESCRIPTION
**Pull request by jiasli**
**Description**  
Fix regression introduced in https://github.com/Azure/azure-cli/pull/13567. Use MSAL-style `authority` to build the `PublicClientApplication`.

**Testing Guide**  
```
az login
az logout
```

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/_identity.py [hunk 1]
```diff
@@ -256,16 +258,15 @@ def _decode_managed_identity_token(self, credential, resource):
         return decoded
 
     def get_user(self, user_or_sp=None):
-        try:
-            return self._msal_app.get_accounts(user_or_sp)
-        except ValueError:
-            pass
+        accounts = self._msal_app.get_accounts(user_or_sp)
```
Comment (qianwens) [inline src/azure-cli-core/azure/cli/core/_identity.py:261]: why remove the try-catch? it throws an exception if the user does not exist which is not handled in the logout function

File: src/azure-cli-core/azure/cli/core/_identity.py [hunk 2]
```diff
@@ -256,16 +258,15 @@ def _decode_managed_identity_token(self, credential, resource):
         return decoded
 
     def get_user(self, user_or_sp=None):
-        try:
-            return self._msal_app.get_accounts(user_or_sp)
-        except ValueError:
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/_identity.py:261]: `ValueError` is thrown due to the incorrect `authority` given to `PublicClientApplication`. If the `username` is not found, `[]` will be return.

**Comment by yonzhan** [comment]
add to S171

File: src/azure-cli-core/azure/cli/core/_identity.py (hunk 1 above)
Comment (jiasli) [reply src/azure-cli-core/azure/cli/core/_identity.py:261]: `ValueError` is thrown due to the incorrect `authority` given to `PublicClientApplication`. If the `username` is not found, `[]` will be return.

**Pull request merged**